### PR TITLE
jQuery.isNumber: Clarify purpose and behavior

### PR DESCRIPTION
**Pull request by gibson042**
Fixes gh-862
Ref jquery/jquery#2781

I held off in case the method gets removed, but that's no reason not to have good documentation while it's still around.

### CODE REVIEW

**Comment by timmywil** [comment]
@gibson042 An updated note is already in the 3.0 docs.

**Comment by gibson042** [comment]
Yes, but it the [update in v3.0.0-docs](https://github.com/jquery/api.jquery.com/commit/4e65b6dfa6c6f2d486572c30dbf5c436b1524030) is independent of gh-862 (which this PR addresses).

**Comment by timmywil** [comment]
It's not completely independent. The notes overlap.

**Comment by timmywil** [comment]
Keep in mind that limiting it to strings and numbers is a change of behavior for 3.0.

**Comment by gibson042** [comment]
Yeah, sorry I missed that. Fixed.

File: entries/jQuery.isNumeric.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <entry type="method" name="jQuery.isNumeric" return="Boolean">
   <title>jQuery.isNumeric()</title>
-  <desc>Determines whether its argument is a number.</desc>
+  <desc>Determines whether its argument represents a Javascript number.</desc>
```
Comment (AurelioDeRosa) [inline entries/jQuery.isNumeric.xml]: JavaScript is misspelled here.

File: entries/jQuery.isNumeric.xml [hunk 2]
```diff
@@ -9,26 +9,31 @@
     </argument>
   </signature>
   <longdesc>
-    <p>The <code>$.isNumeric()</code> method checks whether its argument represents a numeric value. If so, it returns <code>true</code>. Otherwise it returns <code>false</code>. The argument can be of any type.</p>
+    <p>The <code>$.isNumeric()</code> method checks whether a value is a finite number, or is <code>Number</code>-castable to one. If so, it returns <code>true</code>. Otherwise it returns <code>false</code>. The argument can be of any type.</p>
```
Comment (AurelioDeRosa) [inline entries/jQuery.isNumeric.xml]: "or is castable to a Number"? I know what you mean here, but I'm not sure everyone would understand the second part of the sentence.

Comment (gibson042) [inline entries/jQuery.isNumeric.xml]: That's not accurate, though, because finiteness matters. What about "or would be cast to one by `Number`"?

Comment (AurelioDeRosa) [inline entries/jQuery.isNumeric.xml]: Sounds good.

**Comment by gibson042** [comment]
@AurelioDeRosa done.

**Comment by AurelioDeRosa** [comment]
LGTM.

**Pull request closed without merging**